test/integration/daemonset/daemonset_test.go

```diff
@@ -286,6 +332,8 @@ func TestOneNodeDaemonLaunchesPod(t *testing.T) {
 		validateDaemonSetPodsAndMarkReady(podClient, podInformer, 1, t)
 		validateDaemonSetStatus(dsClient, ds.Name, ds.Namespace, 1, t)
 
+		cleanupDaemonSets(t, clientset, ds)
```

Comment:
looks like it could be moved to above after ds is created with `defer`. if it's here, it won't be called if `t.Fatal` is called.